### ISSUE TITLE
Create middleware to read status code from response

### ISSUE THREAD

**Issue by josh-byster** [hack4impact-uiuc/h4i-recruitment#316]
Currently, the user has to set the error code in 2 places and may forget: 

```js
res.status(404).json({
    code: 404,
    message: 'This endpoint must be implemented',
    result: "test"
    success: false
})
```

As part of issue #69 regarding response standardization, we can parse the code from the response object and send that as the status code, or do the opposite and have the code field be added based on the `status` method call.

**Comment by tko22** [comment]
yes, I'd say the best way to do it would be put it in the json and then we parse the code and put it in the response header.

**Comment by josh-byster** [comment]
Yep, I agree. Problem is that the idea of intercepting and modifying a response as it gets sent out is a more difficult undertaking than I originally envisioned. It looks like we'd have to interface directly with the stream API in Node ([example](https://stackoverflow.com/a/15103881/3813411))

**Comment by tko22** [comment]
I think this would be better https://github.com/hack4impact-uiuc/infra-authentication-server/blob/master/src/utils/sendResponse.js#L12
